app/subscribers/super_good/solidus_taxjar/spree/reporting_subscriber.rb

```diff
@@ -13,18 +13,20 @@ module ReportingSubscriber
         event_action :replace_transaction, event_name: :order_recalculated
 
         def report_transaction(event)
-          return unless SuperGood::SolidusTaxjar.configuration.preferred_reporting_enabled
+          order = event.payload[:shipment].order
 
-          SuperGood::SolidusTaxjar::ReportTransactionJob.perform_later(event.payload[:shipment].order)
+          return unless order.completed_at && SuperGood::SolidusTaxjar.configuration.preferred_reporting_enabled_at > order.completed_at
+
+          SuperGood::SolidusTaxjar::ReportTransactionJob.perform_later(order)
         end
 
         def replace_transaction(event)
           order = event.payload[:order]
 
-          return unless SuperGood::SolidusTaxjar.configuration.preferred_reporting_enabled
+          return unless order.completed_at && SuperGood::SolidusTaxjar.configuration.preferred_reporting_enabled_at > order.completed_at
```

Comment:
I don't feel strongly about this, but to make it easier to reason about these guard statements: it _could_ be nice to move them into a private method that describes why we return early:

```suggestion
          return unless reportable?(order)
```

Now, we have a concept of "reportability" with two requirements. And if an end user needed to customize their own reportability requirements they wouldn't have to override the subscribers.